src/cbfs.c

```diff
@@ -0,0 +1,218 @@
+/* SPDX-License-Identifier: GPL-2.0-or-later */
+
+#include "cbfs.h"
+
+#include <unistd.h>
+
+#include <errno.h>
+#include <stdbool.h>
+#include <stdio.h>
+
+#include "boot_data.h"
+#include "utils.h"
+
+#define BOOTORDER_REGION "BOOTORDER"
+#define BOOTORDER_DEF    "bootorder_def"
+#define BOOTORDER_MAP    "bootorder_map"
+
+static FILE *extract(const char *cbfs_tool,
+		     const char *rom_file,
+		     const char *name,
+		     bool is_region)
+{
+	FILE *file;
+	char template[] = "/tmp/cb-order.XXXXXX";
+	char *file_argv[] = {
+		(char *)cbfs_tool, (char *)rom_file, "extract",
```

Comment:
Since this tool is going to be GPLv2, can we copy the files from cbfstool and compile them with this application? Region and raw file access should be enough.

Reply:
Probably, I haven't really considered it. I think `cbfstool` might use some other code from `coreboot`, but it can also be copied.

Reply:
I had some success with this, but then noticed `GPL-2.0-only` in sources from `cbfstool`. We need to support `GPL-3.0` too, because that's the version of modern `libreadline`. By the way, I had to copy thousands of lines (could be reduced though) and it has weird dependencies on parts of `vboot` to do hashing, overall it doesn't really look nice anyway.

Reply:
We do not need hashing actually, so any vboot dependencies should be obsolete. There is only one occurrence of `readline` can we somehow workaround it?

Reply:
Can just implement the prompt manually. Initially I thought that things like file and history completion of `readline` will be useful (e.g., to pick a file), but it doesn't seem to be the case.